src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -24530,6 +24600,436 @@ static void verifyParameters(NamedTypeSymbol typeA)
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeInterfaceImplementation, "set").WithLocation(17, 40),
+                // (21,39): error CS9102: UnscopedRefAttribute cannot be applied to an interface implementation.
+                //     public ref int P1 { [UnscopedRef] get => throw null; } // 4
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeInterfaceImplementation, "get").WithLocation(21, 39),
+                // (22,35): error CS9102: UnscopedRefAttribute cannot be applied to an interface implementation.
+                //     public int P2 { [UnscopedRef] get; set; } // 5
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeInterfaceImplementation, "get").WithLocation(22, 35),
+                // (26,40): error CS9102: UnscopedRefAttribute cannot be applied to an interface implementation.
+                //     public int P2 { get; [UnscopedRef] set; } // 6
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeInterfaceImplementation, "set").WithLocation(26, 40),
+                // (27,35): error CS9102: UnscopedRefAttribute cannot be applied to an interface implementation.
+                //     public int P3 { [UnscopedRef] set { } } // 7
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeInterfaceImplementation, "set").WithLocation(27, 35));
+        }
+
+        [Fact]
+        public void UnscopedRefAttribute_InterfaceImplementation_04()
+        {
+            string source = """
+                using System.Diagnostics.CodeAnalysis;
+                interface I<T>
+                {
+                    ref T F();
+                    ref T P { get; }
+                }
+                struct S : I<int>
+                {
+                    private int _f;
+                    [UnscopedRef] public ref int F() => ref _f;
+                    [UnscopedRef] public ref int P => ref _f;
+                    ref int I<int>.F() => throw null;
+                    ref int I<int>.P => throw null;
+                }
+                """;
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void UnscopedRefAttribute_InterfaceImplementation_05()
+        {
+            string source = """
+                using System.Diagnostics.CodeAnalysis;
+                interface I<T>
+                {
+                    ref T F1();
+                    [UnscopedRef] ref T F2(); // 1
+                }
+                class C1 : I<int>
+                {
+                    private int _f1;
+                    [UnscopedRef] public ref int F1() => ref _f1; // 2
+                    [UnscopedRef] public ref int F2() => ref _f1; // 3
+                }
+                class C2 : I<int>
+                {
+                    private int _f2;
+                    [UnscopedRef] ref int I<int>.F1() => ref _f2; // 4
```

Comment:
> [UnscopedRef] ref int I<int>.F1() => ref _f2; // 4 [](http://example.com/codeflow?start=19&length=51)

Are we testing scenario like this in an interface?

Reply:
`UnscopedRefAttribute_InterfaceImplementation_02`

Reply:
> UnscopedRefAttribute_InterfaceImplementation_02 has interface members marked as [UnscopedRef]. Does that cover the scenario?

Does it test implementing an interface member in an interface?